src/lib/inc/libtimeit/Utils.h

```diff
@@ -26,7 +26,7 @@ time_t getBeginingOfYear(const time_t&);
 time_t getEndOfYear(const time_t&);
 int getDaysInMonth(const time_t &rawtime);
 std::string seconds2hhmm(int64_t s);
-std::string createDurationString(const time_t &from, const time_t &to);
+std::string createDurationString(const time_t &from, const time_t &to, const time_t *next);
```

Comment:
But then it would have to make a `stringstream` from the string returned by the first method, and it would be on the most frequently executed path.  Considering that, do you still suggest the two methods?

Reply:
As this code is, at the moment, only called in the details dialogue and only when a time span is changing value (once a minute) it should not be a problem.

Alternative 1: You can add two strings together with +
Alternative 2: have a private method returning a stringstream that both methods call and then convert into a string.

Optimization would be to update only the timespan row that have actually changed, then you only call this method once every minute. 

BUT: "Premature Optimization Is the Root of All Evil".

Optimize for readability and reliability first. Then monitor performance and do profiling and load tests.

Reply:
One could keep one method and use std::optional for the last parameter.

Reply:
Implementing two functions had a bit of redundancy, for example to get `acrossDays`.  Went back to one function with `std::optional` and then two more functions to call that one function, to facilitate making this feature optional.

Also, updating all rows so far has the advantage when deleting a row the idle time of the row above it gets updated correctly without extra coding effort.

Indeed, premature optimization is or can be a waste of time.  This app is doing just fine as far as I have seen.